features/iotserver-feature/org.wso2.carbon.device.mgt.iot.feature/src/main/resources/jaggeryapps/store/extensions/assets/deviceType/themes/store/js/device-details.js

```diff
@@ -0,0 +1,120 @@
+/*
+ *  Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *
+ */
+
+var modalPopup = ".wr-modalpopup";
+var modalPopupContainer = modalPopup + " .modalpopup-container";
+var modalPopupContent = modalPopup + " .modalpopup-content";
+var body = "body";
+
+/*
+ * set popup maximum height function.
+ */
+function setPopupMaxHeight() {
+    $(modalPopupContent).css('max-height', ($(body).height() - ($(body).height() / 100 * 30)));
+    $(modalPopupContainer).css('margin-top', (-($(modalPopupContainer).height() / 2)));
+}
+
+/*
+ * show popup function.
+ */
+function showPopup() {
+    $(modalPopup).show();
+    setPopupMaxHeight();
+}
+
+/*
+ * hide popup function.
+ */
+function hidePopup() {
+    $(modalPopupContent).html('');
+    $(modalPopup).hide();
+}
+
+/*
+ * DOM ready functions.
+ */
+$(document).ready(function () {
+    attachEvents();
+});
+
+function attachEvents() {
+    /**
+     * Following click function would execute
+     * when a user clicks on "Download" link
+     * on Device Management page in WSO2 DC Console.
+     */
+    $("a.download-link").click(function () {
+        var sketchType = $(this).data("sketchtype");
+        var deviceType = $(this).data("devicetype");
+        var downloadDeviceAPI = "/store/apis/devices/sketch/generate_link";
+        var payload = {"sketchType":sketchType, "deviceType":deviceType};
+        $(modalPopupContent).html($('#download-device-modal-content').html());
+        showPopup();
+        $("a#download-device-download-link").click(function () {
+            console.log("----");
+            invokerUtil.post(
+                downloadDeviceAPI,
+                payload,
+                function (data, textStatus, jqxhr) {
+                    doAction(data);
+                },
+                function (data) {
+                    doAction(data);
+                }
+            );
+        });
+
+        $("a#download-device-cancel-link").click(function () {
+            hidePopup();
+        });
+
+    });
+}
+
+function doAction(data){
+    //if it is saml redirection response
+    if(data.status == null) {
+        document.write(data);
```

Comment:
yes, a comment was added in line 90. when it is a SAML redirection, data.status will be null. However, this file is obsolete since we have dropped ES extended approach.